src/main/kotlin/no/fdk/dataservicecatalog/config/MongoConfig.kt

```diff
@@ -0,0 +1,22 @@
+package no.fdk.dataservicecatalog.config
+
+import com.mongodb.client.MongoClient
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.data.mongodb.config.EnableMongoAuditing
+import org.springframework.data.mongodb.core.MongoOperations
+import org.springframework.data.mongodb.core.MongoTemplate
+
+@Configuration
+@EnableMongoAuditing
+class MongoConfig {
+
+    @Value("\${spring.data.mongodb.database}")
+    lateinit var database: String
+
+    @Bean
+    fun mongoTemplate(mongoClient: MongoClient): MongoOperations {
+        return MongoTemplate(mongoClient, database)
+    }
```

Comment:
Er selv mindre glad i `@Value`, men det fungerer jo, så ikke ta dette som noe krav. Når det er sagt, jeg liker mye bedre `@ConfigurationProperties`:

Om du oppretter feks MongoProperties:
```
@ConfigurationProperties("spring.data.mongodb")
data class MongoProperties (
    val database: String
)
```

Så kan du ta den i bruk som dette:
```suggestion
class MongoConfig(private val mongoProperties: MongoProperties) {

    @Bean
    fun mongoTemplate(mongoClient: MongoClient): MongoOperations {
        return MongoTemplate(mongoClient, mongoProperties.database)
    }
```

Reply:
Eg brukes vanlegvis properties varianten når eg har behov for meir enn ein verdi. Akkurat her er eg usikker på om det blir behov for å hente inn fleire verdier, og då blei det denne varianten.